### PR TITLE
Update Markdownlint to fix issues in linting

### PR DESCRIPTION
**Pull request by kjaymiller**
Currently markdownlint was failing due to configuration issues. 

The following changes have been made

- Run PreCommit on all the files
- Add markdownlint-cli2 to pre-commit
- Replace `markdownlint.json` with `markdownlint.jsonc` which is the default config file type.
- Address failing markdownlint tests with file updates or configuration changes.

[REMINDER]: While we have pre-commit rules we don't want to use pre-commit as an enforcable tool as it can disconnect developers. 

It's more as a convenience to others.

#### Type of Issue

- [x] :bug: (bug)
- [ ] :book: (Documentation)
- [ ] :arrow_up: (Update)
- [ ] :dizzy: (New Feature)
- [ ] :radioactive: (Deprecation)
- [ ] :no_entry_sign: (Removal)
- [ ] :hammer_and_wrench: (Refactor)

#### Changes have been Documented (If Applicable)

- [x] YES - Changes have been documented

#### Changes have been Tested

- [x] YES - Changes have been tested

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+<!-- markdownlint-disable MD041 -->
```
Comment (kjaymiller) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: Supports the accessibility change in our GitHub things which should be an H4 for screen-readers. FMI - #788.

File: .github/dependabot-bot.yml [hunk 2]
```diff
@@ -11,4 +11,4 @@ safe:
   - "render-engine-markdown"
   - "rich"
   - "typer"
-  - "watchdog"
\ No newline at end of file
+  - "watchdog"
```
Comment (kjaymiller) [inline .github/dependabot-bot.yml:14]: ultimately there is a goal to remove dependabot and instead update our environment as a part of pre-commit

File: README.md [hunk 3]
```diff
@@ -1,8 +1,10 @@
+<!-- markdownlint-disable -->
 <!-- ALL-CONTRIBUTORS-BADGE:START - Do not remove or modify this section -->
 
 [![All Contributors](https://img.shields.io/badge/all_contributors-17-orange.svg?style=flat-square)](#contributors-)
 
 <!-- ALL-CONTRIBUTORS-BADGE:END -->
+<!-- markdownlint -->
```
Comment (kjaymiller) [inline README.md:7]: Fixes issue where the All Contributors format was causing the linter to fail.

File: noxfile.py
Comment (kjaymiller) [inline noxfile.py:1]: This is a separate PR that need to be submitted

File: pyproject.toml [hunk 4]
```diff
@@ -40,6 +40,7 @@ dev = [
     "mkdocs-material",
     "mkdocstrings[python]",
     "mypy",
+    "nox",
```
Comment (kjaymiller) [inline pyproject.toml:43]: I thought my `nox` things were already added. this should be a different PR

File: requirements.txt
Comment (kjaymiller) [inline requirements.txt:1]: Not sure why there are so many changes in this.

File: README.md (hunk 3 above)
Comment (john0isaac) [inline README.md:7]: Do we need both lines?
markdownlint-disable and markdonwlint?

File: pyproject.toml (hunk 4 above)
Comment (john0isaac) [reply pyproject.toml:43]: Are you gonna separate them or should we just move forward with this single PR?

File: README.md (hunk 3 above)
Comment (kjaymiller) [reply README.md:7]: yes...

`<!-- markdownlint-disable -->` turns off the linter and it is not turned back on until `<!-- markdownlint -->` is passed.

**Comment by kjaymiller** [comment]
I want to try and separate them but I'm going to be rather slow on this so it may be worthwhile to let this sit for bit and I will try to make the change perhaps in a new PR.
